cterasdk/edge/ssh.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, PublicFormat, NoEncryption
+from .. import config
+from ..lib import FileSystem
+from ..common import Object
+from .base_command import BaseCommand
+
+
+class SSH(BaseCommand):
+    """ Edge Filer SSH daemon APIs """
+
+    def enable(self, public_key=None, exponent=65537, key_size=2048):
+        """
+        Enable the Edge Filer's SSH daemon
+
+        :param str,optional public_key: A PEM-encoded public key in OpenSSH format.
+         If not specified, an RSA key pair will be generated automatically.
+         The PEM-encoded private key will be saved to the default Downloads directory
+        :param int,optional exponent: The public exponent of the new key, defaults to `65537`
+        :param int,optional key_size: The length of the modulus in bits, defaults to `2048`
+        """
+        param = Object()
+
+        private_key = None
+        if not public_key:
+            private_key = rsa.generate_private_key(public_exponent=exponent, key_size=key_size)
+            public_key = private_key.public_key().public_bytes(Encoding.OpenSSH, PublicFormat.OpenSSH)
+
+        param.publicKey = public_key.decode('utf-8')
+
+        if private_key:
```

Comment:
private_key is set only inside the previous if statement context. Why not have this code there? Plus, I'd move the actual save operation into a separate method (just to make the code cleaner)

Reply:
You're right. It bothered me too. Checkout this commit.. I like it better. I'd use the crypto module to:
1) Better certificate management when initializing requests (so we can avoid the warning message we always get)
2) Verify PEM files provided by the user when updating the filer's cert in `edge/ssl.py`